### ISSUE TITLE
Migration to M5 breaks Mautic if mailer password contains special chars

### ISSUE THREAD

**Issue by putzwasser** [mautic/mautic#13236]
### Mautic Version

5.0.x series

### PHP version

8.1

### What browsers are you seeing the problem on?

Not relevant

### What happened?

Updating from M4 to M5 the update script creates the `mailer_dsn` config key for you.

There is a bug in converting special chars properly:

- Given my Amazon SES password looks like this: `sometring/SOMEotherSTRING`.
- The update process/migration will transform it to: `sometring&2FSOMEotherSTRING`, i.e., `urlencode` it.

The resulting `config/local.php` holds now a faulty `mailer_dsn`: 

```php
<?php
$parameters = array(
    // [...]
     'mailer_dsn' => '...://MYUSER:sometring%2FSOMEotherSTRING...',
};
```

This will lead to the following critical PHP error:

```
$ php bin/console

                                                                                                                        
 [WARNING] Some commands could not be registered:                                                                       
                                                                                                                        

In AppKernelProdContainer.php line 11284:
                                                      
  The parameter "2FSOMEotherSTRING" must be defined.  
                                                      
```

Or in the logs:

```
console.CRITICAL: An error occurred while using the console. Message: "The parameter "2FSOMEotherSTRING" must be defined." {"exception":"[object] (Symfony\\Component\\DependencyInjection\\Exception\\InvalidArgumentException(code: 0): The parameter \"2FSOMEotherSTRING\" must be defined. at /.../mautic/docroot/var/cache/prod/Container7PjiqWi/AppKernelProdContainer.php:11284)","message":"The parameter \"2FSOMEotherSTRING\" must be defined."} {"hostname":"mautic.local","pid":24898}
```

The single ampersand in the `mailer_dsn` config key leads Mautic/Symfony to believe there is a parameter that is not defined, hence the error.

The hotfix after updating is commenting out the config key `mailer_dsn` in `local.php` and re-configure your mailer.

### How can we reproduce this issue?

1. Use M4
2. Set up a email transport with a password that contains special chars that need urlencoding
3. Update to M5
4. run `bin/console` or try to access the dashboard.
5. Mautic should fail yielding the mentioned error message

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by dsp76** [comment]
seems somehow be related to https://github.com/mautic/mautic/issues/12256 ?

**Comment by putzwasser** [comment]
I suspect it's only tangential.

I see it as:
- This more about converting the special chars into DSN which fails.
- #12256 and #13228 is about the using the correct encoded special chars from the DSN correctly to authenticate

**Comment by mollux** [comment]
@putzwasser @dsp76 this is indeed a bug, the value is incorrectly stored if any propery of the dsn contains a special character.

I'll create an MR to address this.

**Comment by mollux** [comment]
@putzwasser @dsp76 it's indeed a different issue.

I fixed the migration issue in https://github.com/mautic/mautic/pull/13256, can you test this out?

**Comment by putzwasser** [comment]
@mollux Happy to, but, currently, I have no instance that I could try this out on.

When M5 fully supports SES, etc. I can update another instance. Then I can report back on this.

**Comment by omert37** [comment]
Tried another upgrade attempt, this time to 5.0.2.
Migrated from 4.4.11 to 5.0.2 (successfully with "Success!" message that I now have version 5.0.2) and it would seem that this is my problem as well as my AMAZON SES password contains forward slashes.
After upgrade I get the red "offline" message (see screenshot) with the ""2FXXXXXX" is not defined" (when calling console commands such as cache:clear) message referencing a section in my AMAZON SES password (Just like when I upgraded to 5.0.1 last week).
When I comment out the 'mailer_dsn' line I am able to 'cache:clear' successfully BUT the red offline message (screenshot) is still there when I surf over to mautic from my browser.
![mautic_screenshot](https://github.com/mautic/mautic/assets/33084245/1ef49531-c637-4750-9342-48831846548e)

**Comment by putzwasser** [comment]
If you're certain that `mailer_dsn` doesn't cause the problem you need to check the logs. There's another issue at play, then.

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Forums**. There might be relevant details there:

https://forum.mautic.org/t/upgrade-from-mautic-4-4-11-5-0-2-problem/30602/1

**Comment by putzwasser** [comment]
#13256 fixes this, I guess.
